File: backend/HRManagementSystem/src/Core/HRManagementSystem.Domain/Entities/LeaveRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using HRManagementSystem.Domain.Enums;

namespace HRManagementSystem.Domain.Entities;
public class LeaveRequest: BaseEntity
{
    public int EmployeeId { get; set; }
    public int LeaveTypeId { get; set; }

    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public int TotalDays { get; set; }

    public LeaveStatus Status { get; set; } = LeaveStatus.Pending;

    public int? ReviewedById { get; set; }
    public DateTime? ReviewedAt { get; set; }

    // 🔗 Navigation
    public Employee Employee { get; set; }
    public LeaveType LeaveType { get; set; }
    public Employee? Reviewer { get; set; }
    public ICollection<LeaveApproval>? LeaveApprovals { get; set; }
}

File: frontend/src/app/models/ess/leave-request.ts
export interface LeaveRequest {
  id: number;
  employeeId: number;
  leaveTypeId: number;
  leaveTypeName?: string;
  startDate: string;
  endDate: string;
  totalDays: number;
  status: LeaveStatus;
  reason?: string;
  reviewedById?: number;
  reviewedAt?: string;
  createdAt?: string;
}

export interface CreateLeaveRequest {
  employeeId: number;
  leaveTypeId: number;
  startDate: string;
  endDate: string;
  reason?: string;
}

export enum LeaveStatus {
  Pending = 0,
  Approved = 1,
  Rejected = 2,
}

export const LeaveStatusLabels: Record<LeaveStatus, string> = {
  [LeaveStatus.Pending]: 'Pending',
  [LeaveStatus.Approved]: 'Approved',
  [LeaveStatus.Rejected]: 'Rejected',
};
